cmd/tbcd/README.md

```diff
@@ -1,56 +1,113 @@
+endpoint.**
 
-tbcd is a very minimal bitcoin block downloader and indexer meant for embedding in other applications that require access to bitcoin data (blocks and txes).
+## 🖥️ System Requirements
 
-tbcd requires sufficient disk space for a full download of bitcoin block data on a fast (preferably ssd or better disk.
+`tbcd` requires significant system resources due to its functionality:
 
-tbcd is build with the heminetwork makefile,  To build standalone (requires `go 1.21+`), type:
+| Requirement   | Specification    |
+|---------------|------------------|
+| **CPU Cores** | 4 minimum        |
+| **RAM**       | 8 GiB minimum    |
+| **Disk**      | NVMe recommended |
 
-``` sh
-cd heminetowkr/cmd/tbcd
-go build
-```
+**As of April 18, 2024:**
 
-On some linux systems you may need to increase the number of open files allowed (particularly with slower disks) and the maximum stack size.  If you run into open file or OOM errors, in the shell you are going to run tbcd, run:
+- **`testnet3`** requires approximately 40 GiB of disk space.
+- **`mainnet`** requires over 300 GiB of disk space. <!-- XXX: add exact number here -->
 
-```sh
-ulimit -n 8192
-ulimit -s 8192
-```
+---
+
+## 🛠️ Building `tbcd` From Source
+
+### 🏁 Prerequisites
+
+Ensure Go v1.22 or newer is installed on your system.
+
+### Using Makefile
 
-You can confirm these settings wiht:
+To build `tbcd` using the provided Makefile
 
-```sh
-ulimit -a
+#### 🏁 Prerequisites
+
+Ensure `make` is installed on your system.
+
+#### Build `tbcd` binary
+
+```shell
+cd heminetwork
+
+# Output binary will be written to bin/tbcd or bin/tbcd.exe
+make tbcd
 ```
 
-For a full list of options:
+### Standalone Build
+
+If you prefer not to use the Makefile:
 
-``` sh
-./bin/tbcd --help
+```shell
+cd heminetwork
+
+# Build the binary (output will be tbcd or tbcd.exe)
+go build ./cmd/tbcd/
+
+# Install the binary (output will be in your GOBIN directory)
+go install ./cmd/tbcd/
 ```
 
-You can change the file storage with:
+---
+
+## 🌐 Environment Settings
 
-``` sh
-export TBC_LEVELDB_HOME=/path/to/files
+`tbcd` **checks system limits at startup on supported platforms** to ensure that they are set to values that will allow
+TBC to run without failing.
+
+> [!WARNING]
+> If you see an error similar to the following, you will need to adjust the limits for the number of open files, memory
+> and the maximum stack size on your system to run TBC.
+
+```
+ulimit: memory: limit too low got X, want X
 ```
 
-Specify the network with
+Changing limits is OS-specific, but can usually be done using the `ulimit` command.
 
-``` sh
-export TBC_NETWORK=mainnet
+## ⚙️ Runtime Settings
+
+`tbcd` is **designed to be run in cloud environments**, as such it uses environment variables for runtime settings.
+
+To see a full list of runtime settings, execute `tbcd` with the **`--help`** flag:
+
+```shell
+/path/to/tbcd --help
```

Comment:
since tbcd is written to `bin/` by defualt we can make this:
```
./bin/tbcd --help
```
I think.

Reply:
I think that's better